From e3179a651d6ea8e0c31e55c49a569863b380dac9 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Tue, 28 Jul 2026 20:34:55 -0400
Subject: [PATCH 245/268] atiadlxx: Add stub for
 ADL2_ApplicationProfiles_HitListsX3_Get.

---
 dlls/atiadlxx/atiadlxx.spec   | 2 +-
 dlls/atiadlxx/atiadlxx_main.c | 7 +++++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/dlls/atiadlxx/atiadlxx.spec b/dlls/atiadlxx/atiadlxx.spec
index 120df4fe938..9e9c8767db1 100644
--- a/dlls/atiadlxx/atiadlxx.spec
+++ b/dlls/atiadlxx/atiadlxx.spec
@@ -146,7 +146,7 @@
 @ stub ADL2_ApplicationProfiles_DriverAreaPrivacy_Get
 @ stub ADL2_ApplicationProfiles_GetCustomization
 @ stub ADL2_ApplicationProfiles_HitListsX2_Get
-@ stub ADL2_ApplicationProfiles_HitListsX3_Get
+@ cdecl ADL2_ApplicationProfiles_HitListsX3_Get(ptr long ptr ptr)
 @ stub ADL2_ApplicationProfiles_HitLists_Get
 @ stub ADL2_ApplicationProfiles_ProfileApplicationX2_Assign
 @ stub ADL2_ApplicationProfiles_ProfileApplication_Assign
diff --git a/dlls/atiadlxx/atiadlxx_main.c b/dlls/atiadlxx/atiadlxx_main.c
index 961d3f4260c..9511cd0f4f4 100644
--- a/dlls/atiadlxx/atiadlxx_main.c
+++ b/dlls/atiadlxx/atiadlxx_main.c
@@ -1618,3 +1618,10 @@ int CDECL ADL_Display_SourceContentAttribute_Set(int adapter_index, int display_
 {
     return ADL2_Display_SourceContentAttribute_Set(default_ctx, adapter_index, display_index, attributes);
 }
+
+int CDECL ADL2_ApplicationProfiles_HitListsX3_Get(ADL_CONTEXT_HANDLE ctx, int list_type, int *num_apps, void **list)
+{
+    FIXME("%p %d %p %p stub!\n", ctx, list_type, num_apps, list);
+
+    return ADL_ERR;
+}
-- 
2.55.0

